compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -2016,11 +2016,10 @@ fn assemble_candidates_from_impls<'cx, 'tcx>(
             ImplSource::Builtin(BuiltinImplSource::TraitUpcasting { .. }, _)
             | ImplSource::Builtin(BuiltinImplSource::TupleUnsizing, _) => {
                 // These traits have no associated types.
-                selcx.tcx().dcx().span_delayed_bug(
+                selcx.tcx().dcx().span_bug(
```

Comment:
probably theoretically reachable when manually redefining their lang item :shrug: so we should keep it as a delay span bug